control/control_test.go

```diff
@@ -136,127 +136,133 @@ func getPort() int {
+
+			// Swap plugin that was loaded with a different version of the plugin
+			Convey("Successfull swapping plugins", func() {
+				Convey("Should generate a swapped plugins event", func() {
+					Convey("So first plugin in catalog after swap should have name mock", func() {
+						So(c.PluginCatalog()[0].Name(), ShouldEqual, "mock")
+					})
+					Convey("So swapped plugins event should show loaded plugin name as mock", func() {
+						So(lpe.plugin.LoadedPluginName, ShouldEqual, "mock")
+					})
+					Convey("So swapped plugins event should show loaded plugin version as 1", func() {
+						So(lpe.plugin.LoadedPluginVersion, ShouldEqual, 1)
+					})
+					Convey("So swapped plugins event should show unloaded plugin name as mock", func() {
+						So(lpe.plugin.UnloadedPluginName, ShouldEqual, "mock")
+					})
+					Convey("So swapped plugins event should show unloaded plugin version as 2", func() {
+						So(lpe.plugin.UnloadedPluginVersion, ShouldEqual, 2)
+					})
+					Convey("So swapped plugins event should show plugin type as collector", func() {
+						So(lpe.plugin.PluginType, ShouldEqual, int(plugin.CollectorPluginType))
+					})
+				})
 
-		Convey("Rollback failure returns error", t, func() {
-			lp := c.PluginCatalog()[0]
-			pm := new(MockPluginManagerBadSwap)
-			pm.ExistingPlugin = lp
-			c.pluginManager = pm
+			})
+			Convey("Swap plugin with a different type of plugin", func() {
+				filePath := strings.Replace(fixtures.PluginPathMock2, "snap-plugin-collector-mock2", "snap-plugin-publisher-mock-file", 1)
+				So(filePath, ShouldNotBeEmpty)
+				fileRP, pErr := core.NewRequestedPlugin(fixtures.PluginPathMock1)
+				Convey("Requested publisher plugin should not error", func() {
+					So(pErr, ShouldBeNil)
+					Convey("Swapping collector and publisher plugins", func() {
+						err := c.SwapPlugins(fileRP, c.PluginCatalog()[0])
+						Convey("Should error", func() {
+							So(err, ShouldNotBeNil)
+						})
+					})
+				})
+			})
 
-			mockRP, mErr := core.NewRequestedPlugin(fixtures.PluginPathMock1)
-			So(mErr, ShouldBeNil)
-			err := c.SwapPlugins(mockRP, lp)
-			Convey("So err should be received if rollback fails", func() {
-				So(err, ShouldNotBeNil)
+			//
+			// TODO: Write a proper rollback test as previous test was not testing rollback
+			//
+
+			// Rollback will throw an error if a plugin can not unload
+			Convey("Rollback failure returns error", func() {
+				lp := c.PluginCatalog()[0]
+				pm := new(MockPluginManagerBadSwap)
+				pm.ExistingPlugin = lp
+				c.pluginManager = pm
+
+				mockRP, mErr := core.NewRequestedPlugin(fixtures.PluginPathMock1)
+				So(mErr, ShouldBeNil)
+				err := c.SwapPlugins(mockRP, lp)
+				Convey("So err should be received if rollback fails", func() {
+					So(err, ShouldNotBeNil)
+				})
 			})
+			c.Stop()
+			time.Sleep(100 * time.Millisecond)
 		})
-
-		c.Stop()
-		time.Sleep(100 * time.Millisecond)
+	} else {
+		fmt.Printf("SNAP_PATH not set. Cannot test swapping plugins.\n")
```

Comment:
@lmroz, @IRCody, @jcooklin - simply question: **What will be better to do when SNAP_PATH is not set** - skip test (option A) or fail (option B)?

**Option A:**
```
       if fixtures.SnapPath == "" {
              t.Log("SNAP_PATH not set. Cannot test swapping plugins.\n")
              t.SkipNow()
        }
```
results:
![image](https://cloud.githubusercontent.com/assets/11335874/20008769/1ae04cae-a2a2-11e6-95c9-36eb9a232a73.png)

**Option B**:
```
 if fixtures.SnapPath == "" {
                t.Error("SNAP_PATH not set. Cannot test swapping plugins.\n")
        }
```
results:
![image](https://cloud.githubusercontent.com/assets/11335874/20008799/3d4f5500-a2a2-11e6-9c6f-daa8cb9306b6.png)

In my opinion, option B is better, because:
- these tests are more like basic tests (testing loading, unloading, swapping, etc.), so passed Travis tests should equals that all of them were executed and passed. They should not be just skipped.
- lack of SNAP_PATH means that [scripts/test.sh](https://github.com/intelsdi-x/snap/blob/master/scripts/test.sh#L32) is wrong 

Let me know do You agree with me. Thanks.